orangecontrib/spectroscopy/io/neaspec.py

```diff
@@ -11,195 +11,285 @@
+            # For old neaspec filenformat
+            data_reader = readers.NeaFileLegacyReader(self.filename)
+            data, _ = data_reader.read()
+            channels = list(data.keys())[3:]
+        elif self.filename.endswith(".txt"):
+            data_reader = readers.NeaHeaderReader(self.filename)
+            channels, _ = data_reader.read()
+            channels.insert(0, "All")
+            channels = [
+                c
+                for c in channels
+                if c not in ("Row", "Column", "Run", "Omega", "Wavenumber", "Depth")
+            ]
+        else:
+            channels = []
+
+        return channels
+
+    @staticmethod
+    def detect_image_signaltype(filename):
+        # Copied from NeaImageGSF
+        # We will need to change this when neaspec comes up with new channel names
+        channel_strings = ["M(.?)A", "M(.?)P", "O(.?)A", "O(.?)P", "Z C", "Z raw"]
+        channel_name = None
+
+        for pattern in channel_strings:
+            if re.search(pattern, filename) is not None:
+                channel_name = re.search(pattern, filename)[0]
+
+        if channel_name is None:
+            signal_type = "Topography"
+        elif "P" in channel_name:
+            signal_type = "Phase"
+        elif "A" in channel_name:
+            signal_type = "Amplitude"
+        else:
+            signal_type = "Topography"
+
+        return signal_type
+
+    def read_spectra(self):
+        if self.filename.endswith(".gsf"):
+            # In case of GSF images the wavelength is
+            # in a separate file or in the filenames
+            wn = readers.get_wl_from_filename(self.filename)
+            if wn is None:
+                wn = readers.get_wl_from_infofile(self.filename)
+            if wn is None:
+                wn = 1.0
+
+            X, XRr, YRr = reader_gsf(self.filename)
+            features, final_data, meta_data = _spectra_from_image(
+                X, np.array([wn]), XRr, YRr
+            )
+            signal_type = self.detect_image_signaltype(self.filename)
+            meta_data.attributes["measurement.signaltype"] = signal_type
+
+            # TODO: add all the meta info here from the Gwyddion header
+            return features, final_data, meta_data
+
+        if self.filename.endswith(".nea"):
+            # For old neaspec format
+            data_reader = readers.NeaFileLegacyReader(self.filename)
+            data, measparams = data_reader.read()
+        else:
+            # Current fileformat
+            data_reader = readers.NeaSpectralReader(self.filename)
+            data, measparams = data_reader.read()
+
+        if self.sheet:
+            chn = self.sheet
+        else:
+            chn = self.sheets[0]
+            self.sheet = chn
+
+        if chn == "All":
+            N_chn = len(self.sheets) - 1
+        else:
+            N_chn = 1
+
+        # There are strored in the measparams
+        Max_row = int(np.max(data["Row"]) + 1)
```

Comment:
Also here, why int() and +1?

If this was an int value anyway, and +1 is added to get a count, then name is wrong. Then it should be `n_rows` or something similar.